src/main/java/apartment/domain/Result.java

```diff
@@ -0,0 +1,20 @@
+package apartment.domain;
+
+import java.util.List;
+
+public class Result<T> extends Response {
+
+    private T payload;
+
+    public T getPayload() {
+        return payload;
+    }
+
+    public void setPayload(T payload) {
+        this.payload = payload;
+    }
+
+    public void setMessage(List<String> messages) {
```

Comment:
```suggestion
    public void setMessages(List<String> messages) {
```

Reply:
When I saw this setter early in your code it confused because you were passing a collection of messages into a setting that referred to a single message.

Reply:
Also, I think once you sort your Result type for the above `update()` method, you can remove this setter.